torch/fx/experimental/proxy_tensor.py

```diff
@@ -479,21 +473,24 @@ def enable(self, b):
     def __sym_dispatch__(self, func, types, args, kwargs):
         if not self.enable_tracing:
             return func(*args, **kwargs)
-        p_args, p_kwargs = pytree.tree_map_only(
-            (PySymInt, PySymFloat),
-            lambda s: get_proxy_slot(s, self.tracer) if s.constant is None else s.constant,
-            (args, kwargs)
+        # For speed, we assume there are no nested data structures
```

Comment:
how much information do we have about the schema of these functions - are they always aten, so they have a schema?  would that let us codegen c++ that knew what structure to expect the input args and thus avoid treemaps?

Reply:
So, SymInt is special because they have no schema; but for tensor ops, yes, this is an optimization I considered. It's a little tricky though because you need to make this info available from Python and there isn't a clean API for doing so.

Reply:
ok, well at least we have some ideas on the back burner if we need to go deeper.